<launch>

  <!-- args that can be set from the command line or a default will be used -->
  <arg name="deformation_model_path" default="/home/mullis/drapebot_dataset/m28_demonstration/F20/models/regression/Densenet121_0.0001_1/epoch39.pth"/>
  <arg name="dofs" default="4"/>
  <arg name="half_range_def" default="[0.12, 0.365, 0.12, 0.4188790204786391]"/>
  <arg name="mid_def" default="[0.0, 0.905, 0.0, 0.0]"/>
  <arg name="segmentation_method" default="NN"/>
  <arg name="segmentation_model_path" default="/home/mullis/ros2_projects/deformable_ws/src/soft_material_deformation_estimation_ros2/models/segmentation/UNet_resnet18_1e-05_0.pth"/>


  <node pkg="soft_material_deformation_estimation_ros2" exec="deformation_estimation" name="deformation_estimation" output="screen">
    <param name="deformation_model_path" value="$(var deformation_model_path)"/>
    <param name="dofs" value="4"/>
    <param name="half_range_def" value="$(var half_range_def)"/>
    <param name="mid_def" value="$(var mid_def)"/>
    <param name="segmentation_method" value="$(var segmentation_method)"/>
    <param name="segmentation_model_path" value="$(var segmentation_model_path)"/>

  </node>
</launch>